### PR TITLE
Split the e2e and vr tests

### LINKED ISSUE #2041: Make e2e and visual regression Playwright tests run in parallel
## Problem

Currently, the Nuxt Playwright tests step of the CI workflow builds the Nuxt app and then runs the e2e tests, followed by the visual regression tests. This takes a long time, between 15 and 20 minutes.

## Description

We could split the Playwright tests into two parallel actions: e2e and visual regression. This would make them faster. However, if we simply split the two, we would need to build the Nuxt app twice. 

@WordPress/openverse-frontend, @sarayourfriend, is there a way to build the Nuxt app once and re-use it for both e2e and visual regression Playwright tests?

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2041 by @obulat

## Description

This PR splits the Playwright tests in CI into two separate steps: VR and e2e. This makes the tests faster, but the drawback is that the CI now builds the Nuxt app twice.

The CI now lasts less than 13 minutes. I was surprised to learn that e2e tests take longer than the VR tests.

This PR also adds some changes to stop the flakiness that can currently be seen on main:
- it manually sets the search input height for screens wider than `md` to 3rem (48px) to make sure that the height of the input does not cause flakiness anymore. It's a hotfix that I think should be OK for now because we are not going to be use the old header soon.
- it adds the cookies that affect the layout to the context in the pages Playwright tests.

## Testing Instructions

The CI should pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,9 +131,10 @@ jobs:
   playwright-test-failure-comment:
     name: Share Playwright test debugging instructions
     runs-on: ubuntu-latest
-    if: always() && github.ref != 'refs/heads/main' && (needs.nuxt-playwright.result == 'failure' || needs.storybook-playwright.result == 'failure')
+    if: always() && github.ref != 'refs/heads/main' && (needs.nuxt-playwright-vr.result == 'failure' || needs.nuxt-playwright-e2e.result == 'failure' ||needs.storybook-playwright.result == 'failure')
     needs:
-      - nuxt-playwright
+      - nuxt-playwright-e2e
+      - nuxt-playwright-vr
       - storybook-playwright
```
Comment (dhruvkb) [inline .github/workflows/ci.yml:138]: In the interest of clarity
```suggestion
      - storybook-playwright-vr
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -73,22 +73,40 @@ jobs:
...
-        run: pnpm test:playwright --workers=2
+      - name: Run Nuxt Playwright e2e tests
+        run: pnpm test:playwright e2e --workers=2
 
       - uses: actions/upload-artifact@v3
         if: failure()
         with:
-          name: nuxt-test-results
+          name: nuxt-e2e-test-results
+          path: test-results-e2e
```
Comment (dhruvkb) [inline .github/workflows/ci.yml:92]: This should still be `test-results/` unless we change the `outputDir` in the Playwright config.

```suggestion
          path: test-results
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
If we're building the Nuxt server twice (which is a long process) I don't expect a significant gain in CI time. But I still like the idea of splitting E2E from VR and support this PR.

**Comment by obulat** [comment]
> If we're building the Nuxt server twice (which is a long process) I don't expect a significant gain in CI time. But I still like the idea of splitting E2E from VR and support this PR.

It appears that the build process for the Nuxt server in CI lasts around 3 minutes, and because both tests don't run in sequence, the tests last around 11-13 minutes now:
Screenshot 2022-12-13 at 3 46 27 PM

**Comment by dhruvkb** [comment]
So we went from ~18 to ~12 (-33%) and can add another 3 minutes of VR tests without affecting future CI times. I will take back my words, that's a pretty big improvement!

**Review by dhruvkb — APPROVED** [review]

File: src/components/VBanner/VBanners.vue [hunk 3]
```diff
@@ -37,12 +37,17 @@ export default defineComponent({
...
+      uiStore.dismissBanner(bannerKey)
+    }
+
     return {
       translationBannerId,
       shouldShowMigrationBanner,
       shouldShowTranslationBanner,
 
-      dismissBanner: uiStore.dismissBanner,
+      dismissBanner,
```
Comment (zackkrida) [inline src/components/VBanner/VBanners.vue:50]: Did this change have a noticeable effect on the banner behavior? It seems like a welcome change regardless.

**Review by zackkrida — APPROVED** [review]
The small bugs fixed here and the logic to set cookies in the visual regression tests appears to work well. Just curious, were those secondary bugs identified as part of this PR or earlier? I'd prefer to see those changes decoupled from the main goal of the PR in the future.

**Pull request merged**